### PR TITLE
Improvement of create table widget

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Implements #214

##### Description of changes
- Make widget remembers the data for the domain in the context
- Increase the initial table size
- Allow mixed numbers and string in the same column
- Support for the time variable
- Fix bug: data removed when line edit opened
- Tests
- Code cleanup

#### TODO:
- Documentation

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/prototypes/widgets/owcreatetable.py [hunk 1]
```diff
@@ -160,21 +220,44 @@ class Outputs:
...
+        transform_err = Msg("Data does not fit to domain")
+
+    settingsHandler = PerfectDomainContextHandler()
 
-    n_rows = Setting(1)
-    n_columns = Setting(1)
+    n_rows = Setting(len(DEFAULT_DATA))
+    n_columns = Setting(len(DEFAULT_DATA[0]))
     auto_commit = Setting(True)
+    # since data is a small (at most 20x20) table we can afford to store it
```
Comment (ajdapretnar) [inline orangecontrib/prototypes/widgets/owcreatetable.py:230]: When input domain is larger than 20 variables, that would not be the case. The number of columns is arbitrarily large. The number of rows, however, is fixed to 20. Is that still a limitation is this case?

**Comment by ajdapretnar** [comment]
This works quite well and I'd be happy to merge. One small thing. When I enter custom data, then feed the widget a domain (say iris), then remove the connection, I would expect to have the previous custom data waiting for me. It happens for the data based on a domain, so I see no reason not to have that for custom data, too.

File: orangecontrib/prototypes/widgets/owcreatetable.py (hunk 1 above)
Comment (PrimozGodec) [reply orangecontrib/prototypes/widgets/owcreatetable.py:230]: When the input domain is larger we allow more columns. I think it is OK, nobody would have a patience to fill 100 columns and 20 rowns manually.

**Comment by PrimozGodec** [comment]
> This works quite well and I'd be happy to merge. One small thing. When I enter custom data, then feed the widget a domain (say iris), then remove the connection, I would expect to have the previous custom data waiting for me. It happens for the data based on a domain, so I see no reason not to have that for custom data, too.

Yep, this is the only reason why it is still WIP. I will probably implement that tommorrow.

**Comment by PrimozGodec** [comment]
> This works quite well and I'd be happy to merge. One small thing. When I enter custom data, then feed the widget a domain (say iris), then remove the connection, I would expect to have the previous custom data waiting for me. It happens for the data based on a domain, so I see no reason not to have that for custom data, too.

It should work now. :) from my side widget can be merged if everything is OK

**Comment by ajdapretnar** [comment]
I am merging, since it is a prototype. There is one thing that need to be fixed, but I will open a separate issue for this.

**Pull request merged**